test/plugin.test.js

```diff
@@ -349,3 +349,49 @@ test('add hooks after route declaration', t => {
     })
   })
 })
+
+test('nested plugins', t => {
+  t.plan(5)
+
+  const fastify = Fastify()
+
+  t.tearDown(fastify.close.bind(fastify))
+
+  fastify.register(function (fastify, opts, next) {
+    fastify.register((fastify, opts, next) => {
+      fastify.get('/', function (req, reply) {
+        reply.send('I am child 1')
+      })
+      next()
+    }, { prefix: '/child1' })
+
+    fastify.register((fastify, opts, next) => {
+      fastify.get('/', function (req, reply) {
+        reply.send('I am child 2')
+      })
+      next()
+    }, { prefix: '/child2' })
+
+    next()
+  })
```

Comment:
@delvedor this test should be the one that you referring to.

Reply:
Yes, it is!
I would like to add another prefix (`/parent`) to be super sure about the parent's prefix as well.